X10D.Unity/src/Numerics/QuaternionExtensions.cs

```diff
@@ -18,7 +19,7 @@ public static class QuaternionExtensions
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static System.Numerics.Quaternion ToSystemQuaternion(this Quaternion quaternion)
     {
-        return new System.Numerics.Quaternion(quaternion.x, quaternion.y, quaternion.z, quaternion.w);
+        return UnsafeUtility.As<Quaternion, System.Numerics.Quaternion>(ref quaternion);
```

Comment:
This is good. I was unaware that Unity had its own version of `Unsafe` to allow you to reinterpret data this way.